mbed_targets/_internal/target_attribute_hierarchy_parsers/accumulating_attribute_parser.py

```diff
@@ -13,18 +13,31 @@
 from typing import Dict, List, Any, Deque
 
 ACCUMULATING_ATTRIBUTES = ["extra_labels", "macros", "device_has", "features", "components"]
-ALL_ACCUMULATING_ATTRIBUTES = ACCUMULATING_ATTRIBUTES + [
-    "extra_labels_remove",
-    "extra_labels_add",
-    "macros_remove",
-    "macros_add",
-    "device_has_remove",
-    "device_has_add",
-    "features_remove",
-    "features_add",
-    "components_remove",
-    "components_add",
-]
+ACCUMULATING_ATTRIBUTES_MODIFIERS = {"add": "_add", "remove": "_remove"}
+
+
+def _modifying_attribute_name(attribute: str, modifying_action: str) -> str:
+    return f"{attribute}{ACCUMULATING_ATTRIBUTES_MODIFIERS[modifying_action]}"
+
+
+def _list_all_accumulating_attributes_and_modifiers() -> List[str]:
+    """Provides a list of all accumulating attributes and their modifying attributes.
+
+    For example, the accumulating attribute "extra_labels" can be modified by "_add" and
+    "_remove" to make "extra_labels_add" and "extra_labels_remove".
+
+    Returns:
+        A list of all the accumulating attributes and their modifiers.
+    """
+    all_accumulating_attributes = ACCUMULATING_ATTRIBUTES.copy()
+    for modifier_action in ACCUMULATING_ATTRIBUTES_MODIFIERS.keys():
+        all_accumulating_attributes.extend(
+            [_modifying_attribute_name(attribute, modifier_action) for attribute in ACCUMULATING_ATTRIBUTES]
+        )
+    return all_accumulating_attributes
+
+
+ALL_ACCUMULATING_ATTRIBUTES = _list_all_accumulating_attributes_and_modifiers()
```

Comment:
I am not sure about this. I don't like that I have to put the private functions above this in the file, but I also don't want to just provide a function that recreates the list over and over again unnecessarily. I also want to be able to pull the constant `ALL_ACCUMULATING_ATTRIBUTES` into the overriding attributes parser to create `NON_OVERRIDING_ATTRIBUTES`.  🤔  Maybe I am taking the wrong approach. Any suggestions welcome.

Reply:
Should we make this a class and just create the `all_accumulating_attributes` as an instance variable in the `__init__` method?

Reply:
I'm not sure that would much help. I want to be able to access it from other files without having to initialise anything.